src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -164,6 +182,28 @@ private enum Flags : byte
             }
         }
 
+        internal SynthesizedBackingFieldSymbol CreateBackingField(bool isCreatedForFieldKeyword)
+        {
+            Debug.Assert(!IsIndexer);
+            if (_lazyBackingFieldSymbol is null)
+            {
+                var backingField = new SynthesizedBackingFieldSymbol(this,
+                                              GeneratedNames.MakeBackingFieldName(_name),
+                                              isReadOnly: (HasGetAccessor && !HasSetAccessor) || IsInitOnly,
+                                              this.IsStatic,
+                                              hasInitializer: (_propertyFlags & Flags.HasInitializer) != 0,
+                                              isCreatedForfieldKeyword: isCreatedForFieldKeyword);
+                InterlockedOperations.Initialize(ref _lazyBackingFieldSymbol, backingField);
+            }
+
+            if (isCreatedForFieldKeyword) // If not created for field keyword, the flag is set in SourcePropertySymbolBase ctor.
+            {
+                _propertyFlags |= Flags.IsAutoProperty; // PROTOTYPE(semi-auto-props): Revise precisely when we need to set this.
```

Comment:
>Flags.IsAutoProperty [](http://example.com/codeflow?start=34&length=20)

Why do we need to set this flag? What difference does this make?

Reply:
> > Flags.IsAutoProperty
> 
> Why do we need to set this flag? What difference does this make?

I haven't confirmed, but I think this is a code path that can be affected.

https://github.com/dotnet/roslyn/blob/e3154ba0793003b9f59d0b439d0a5c57e7ea861b/src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs#L674

Tracking adding more tests and tracking the correctness of the line is driven by the prototype comment currently.

Reply:
I validated that this line is needed for the following to compile correctly:

```csharp
public string P { get => field; set => field = value; } = "Hello";
```

Otherwise, we get an `error CS8050: Only auto-implemented properties can have initializers.`

I can see the flag isn't actually needed. `IsAutoProperty` can be implemented via `_lazyBackingFieldSymbol is not null` instead of checking the flag. Let me try that and see if anything will break.

Reply:
> I validated that this line is needed for the following to compile correctly ...

I think I already commented on this issue on a separate thread. I understand that this is the way to get the scenario working. However, I am not yet convinced that this is the right way to get it working. I think we have the following options:
- Figure out the better place to perform the check, when we can be sure that the normal binding process happened for the properties.
- Temporarily disable the check for properties that we marked as possibly using the ```field``` keyword. Add a PROTOTYPE comment about that. Disable relevant failing tests for scenarios when the identifier is not a keyword with a PROTOTYPE comment as well. Look for a better solution in a separate PR.